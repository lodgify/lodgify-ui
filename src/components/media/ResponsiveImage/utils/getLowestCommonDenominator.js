/**
 * @param  {number} firstNumber
 * @param  {number} secondNumber
 * @return {number}
 */
export const getLowestCommonDenominator = (firstNumber, secondNumber) =>
  secondNumber > 0
    ? getLowestCommonDenominator(secondNumber, firstNumber % secondNumber)
    : firstNumber;
