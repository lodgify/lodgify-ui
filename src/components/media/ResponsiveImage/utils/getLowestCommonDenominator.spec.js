import { getLowestCommonDenominator } from './getLowestCommonDenominator';

describe('`getLowestCommonDenominator`', () => {
  it('should return the right value', () => {
    const actual = getLowestCommonDenominator(1, 4);

    expect(actual).toMatchSnapshot();
  });
});
