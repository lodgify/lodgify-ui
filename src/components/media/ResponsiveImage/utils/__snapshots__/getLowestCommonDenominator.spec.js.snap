// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`getLowestCommonDenominator\` should return the right value 1`] = `1`;
